plugins/extractit/icon_extractit/util/extractor.py

```diff
@@ -12,11 +13,108 @@
+    :type cell_number_format: str
+
+    :return: Returns formatted for datetime cell number format
+    :rtype: str
+    """
+
+    return (
+        cell_number_format.replace("mm", "%m")
+        .replace("dd", "%d")
+        .replace("yyyy", "%Y")
+        .replace("yy", "%y")
+        .replace('"', "")
+        .lstrip("0")
+    )
+
+
+def _get_non_empty_cells(input_unzipped_file: zipfile.ZipFile) -> List[str]:
+    """Gets all the cell names (i.e. A1) when it's value is not empty
+
+    :param input_unzipped_file: Unzipped file containing all the xml files
+    :type input_unzipped_file: zipfile.ZipFile
+
+    :return: Returns list containing non-empty cell names
+    :rtype: List[str]
+    """
+
+    output_cell_values = []
+    for file_ in input_unzipped_file.infolist():
+        if "worksheet" in file_.filename and file_.filename.endswith("xml"):
+            output_cell_values += regex.findall(
+                r'r="[A-Z]+\d+"', input_unzipped_file.read(file_).decode(DEFAULT_ENCODING)
+            )
+    return list({cell[3:].strip('"') for cell in output_cell_values})
+
+
+def _extract_dates_from_cells(input_sheet: str, cells_with_values: List[str]) -> List[str]:
+    """Extracts the dates from all non empty cells that are formatted as Date and Time cells
+    with the date format defined in the xlsx sheet
+
+    :param input_sheet: Name of the sheet from where the data will be obtained
+    :type input_sheet: str
+    :param cells_with_values: List that contains all the non-empty cell names
+    :type cells_with_values: List[str]
+
+    :return: Returns list containing Date Time values from the input cells
+    :rtype: List[str]
+    """
+
+    output_dates = []
+    for cell in cells_with_values:
+        cell_value = input_sheet[cell].value
+        number_format = input_sheet[cell].number_format
+        if cell_value and isinstance(cell_value, datetime):
+            strf_date = cell_value.strftime(_get_cell_number_format(input_sheet[cell].number_format))
+            if number_format.count("m") == 1 and number_format.count("d") == 1:
+                strf_date = cell_value.strftime(strf_date.replace("m", "%m").replace("d", "%d"))
+            output_dates.append(strf_date)
+    return output_dates
+
+
+def _extract_dates_from_xlsx(input_file: Any, unzipped_file: zipfile.ZipFile) -> str:
+    """Extracts all the dates from .xlsx file
+
+    :param input_file: Opened in buffer xlsx file
+    :type input_file: Any
+    :param unzipped_file: Unzipped file to date extract from
+    :type unzipped_file: zipfile.ZipFile
+
+    :return: Str containing all the ęxtracted date and time values separated by space character
+    :rtype: str
+    """
+
+    try:
+        workbook = openpyxl.load_workbook(input_file)
+        sheet_names = workbook.sheetnames
+        output_list_of_dates = []
+        for sheet_name in sheet_names:
+            sheet = workbook[sheet_name]
+            output_list_of_dates += _extract_dates_from_cells(
```

Comment:
Thinking, could the use of these two method be simplified to use openpyxl iter_rows method? Unsure if it is faster.

```
def _get_dates_from_cells(sheet:openpyxl.workbook.workbook.Worksheet) -> List[str]:
    output_dates = []
    for row in sheet.iter_rows():
        for cell in row:
            if cell.value and isinstance(cell.value, datetime):
                strf_date = cell.value.strftime(_get_cell_number_format(cell.number_format))
                if cell.number_format.count("m") == 1 and cell.number_format.count("d") == 1:
                    strf_date = cell.value.strftime(strf_date.replace("m", "%m").replace("d", "%d"))
                output_dates.append(strf_date)
    return output_dates
```

Reply:
This is much better that I got there! Thank you so much! 😄 it works as fast as before!